### ISSUE TITLE
[3.0.0] Text Report broken

### ISSUE THREAD

**Issue by ek9** [leanphp/behat-code-coverage#12]
Text report seems to be broken on `3.0.0`

**Comment by stevepetcu** [comment]
Hi! What is this bug about? I've been trying to use this extension today, but it's generating empty reports (tested with `php` and `html`).

Running tests "locally", inside a docker container, with XDebug installed and on (the coverage works for phpunit and phpspec - with leanphp).

**Comment by nedoc1990** [comment]
@stevepetcu Any workaround? I keep getting empty reports.

**Comment by golovetska** [comment]
Having the same issue with empty reports (doesn't matter what format I choose). ANy ideas how to solve it?

**Comment by golovetska** [comment]
@nedoc1990 I've managed to get it working with some small changes. You can check my commit if you are interested: https://github.com/golovetska/behat-code-coverage/commit/b883fdc721b68734ca8f000e00d7aa56c99a2906

**Comment by nedoc1990** [comment]
@golovetska thanks for replying! I checked your commit and applied the changes, still not working :(

I'm using symfony 3.3.13, behat 3.3.1, and the default configuration of [behat-codecoverage-extension](https://github.com/leanphp/behat-code-coverage)

**Comment by golovetska** [comment]
@nedoc1990  This is my config file, though it doesn't look that much different than a default one, anyway will leave it here:

```
    LeanPHP\Behat\CodeCoverage\Extension:
      auth:       ~
      drivers:
        - local
      filter:
        forceCoversAnnotation:                true
        mapTestClassNameToCoveredClassName:   true
        whitelist:
          addUncoveredFilesFromWhitelist:     true
          processUncoveredFilesFromWhitelist: true
          include:
            directories:
              'src':
                prefix: 'src'
              'tests':
                prefix: 'src'
          exclude:
            directories:
              'vendor':
                prefix: 'vendor'
      report:
        format:   html
        options:
          target: build/behat-coverage
```
I am using behat 3.2 and some components from symfony 3 btw.

**Comment by nedoc1990** [comment]
@golovetska don't exactly know why but i'm getting nice html reports again. seems like the filter section did the trick. thanks again.

**Comment by stevepetcu** [comment]
I may give it a shot next month. Sorry - I'm busy with moving to the UK and
I had to totally ignore some stuff like this.

In the meantime, I went with full coverage from PHPSpec and PHPUnit, even
though it meant double checking some stuff that's already covered with
Behat/being an overkill. Used sebastianbergmann/php-code-coverage to merge
the coverage reports from those tools - I think this also helps keep the
original purpose of Behat testing intact and it makes the actual coverage
from Behat tests far less important.

**Comment by golovetska** [comment]
@stevepetcu Do you maybe have published a script which you use to merge code coverage from different sources? do you merge html format?
I was checking sebastianbergmann/php-code-coverage before for exactly the same purpose. As I understood it was possible to merge only coverage in clover format but I might be wrong. 

Good luck with the move to the UK!

**Comment by stevepetcu** [comment]
@golovetska As a matter of fact, I have such a script, albeit not published in a public repository yet. Here you go:

```bash
#!/bin/bash
# get the path of the coverage directory, relative to this script
DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/../.coverage/"

# run all the PHPUnit and PHPSpec tests, and merge the coverage.
printf "Running PHPUnit test suites...\n\n"
vendor/bin/phpunit --coverage-php $DIR/.cov/coverage_phpunit.cov

printf "\nRunning PHPSpec test suites...\n\n"
vendor/bin/phpspec run

printf "\nMerging results in '<project_root>/.coverage'...\n\n"
vendor/bin/phpcov merge $DIR/.cov --html $DIR

printf "\nDone!\n\n Open '<project_root>/.coverage/index.html' in a browser to see the results.\n\n"

exit 0
```

You will indeed, generate the coverage reports in clover format, then merge them and generate an HTML report using php-code-coverage. I have a `.coverage/` folder in the root of my project (which is committed, b/c the tools are only able to create one level of folders, e.g., they can create `.cov/`. but not `.coverage/.cov`) for doing this stuff.

**Comment by golovetska** [comment]
@stevepetcu Thanks a lot! That's exactly what I needed.

**Comment by ek9** [comment]
I will have a look in getting this merged during this weekend, thank you everyone for providing detailed information.

**Comment by ivan1986** [comment]
also
```
directories:
  'src':
    prefix: ''
  'tests':
    prefix: ''
```
fix this

**Comment by ek9** [comment]
I'm sorry but I have had delays getting this issue sorted. Will make sure this is sorted in the coming week.

**Comment by dannylewis-sheffield** [comment]
Created pull request https://github.com/leanphp/behat-code-coverage/pull/23 with patch provided by @golovetska  Thanks @golovetska !

**Comment by ek9** [comment]
Ok now to clear some things up:

1. This issue was originally made because Text report seem to be not functioning. It still is not functioning.
2. @stevepetcu @nedoc1990 @golovetska @dannylewis-sheffield @ivan1986 - the problem discussed in this issue that you were having is a different issue. There was a bug that `prefix` would default to `src` if not specified, this would make the whitelisting not work correctly and prevent  code generation to work in general. I have made a separate issue ( #28 ) and will shortly merge a PR to fix it ( #29 ). Besides the fixes, it will also include updated `README` and `behat.yml.dist` so setting up is less complicated.

Let's leave this issue for what it originally was - getting Text Report generation fixed.

**Comment by ek9** [comment]
Now for the issue of the actual Text Report - the report would be generated, but the extension would not provide any output to the console as it would not be printed. I believe this bug was introduced in the original extension.

Issuing a PR to fix it shortly.
